include/gsl/span

```diff
@@ -351,19 +381,20 @@ namespace details
 } // namespace details
 
 // [span], class template span
-template <class ElementType, std::ptrdiff_t Extent>
+template <class ElementType, std::size_t Extent>
 class span
 {
 public:
     // constants and types
     using element_type = ElementType;
     using value_type = std::remove_cv_t<ElementType>;
-    using index_type = std::ptrdiff_t;
+    using index_type = std::size_t;
```

Comment:
`std::span` has `size_type` but not `index_type`. You should consider adding `size_type` for interface compatibility, and possibly consider marking `index_type` as deprecated.